### PR TITLE
[+] Project : Add hooks on Object init

### PR DESCRIPTION
**Pull request by Shagshag**
For example, it gives possibility to add properties to the object.

``` php
class MyModule extends Module
{
    public function hookActionObjectProductInit($params)
    {
        $property = $this->getProductProperty($params['object']->id);
        $params['object']->my_property = $property;
    }

    public function hookActionObjectProductUpdateAfter($params)
    {
        $this->setProductProperty($params['object']->id, $params['object']->my_property);
    }

    public function hookActionObjectProductAddAfter($params)
    {
        $this->setProductProperty($params['object']->id, $params['object']->my_property);
    }

    public function hookActionObjectProductDeleteAfter($params)
    {
        $this->deleteProductProperty($params['object']->id);
    }
}
```

### CODE REVIEW

File: classes/ObjectModel.php [hunk 1]
```diff
@@ -232,6 +232,11 @@ public function __construct($id = null, $id_lang = null, $id_shop = null)
             $entity_mapper = Adapter_ServiceLocator::get("Adapter_EntityMapper");
             $entity_mapper->load($id, $id_lang, $this, $this->def, $this->id_shop, self::$cache_objects);
         }
+	
+	// @hook actionObject*Init
+	Hook::exec('actionObjectInit', array('object' => $this));
```
Comment (gRoussac) [inline classes/ObjectModel.php]: Hi,

Thank you for you Pr. I believe you have a different indentation, are you using spaces or tabs ?

Thank you. Regards

**Comment by guestisp** [comment]
This could lead to custom customer's fields, by creating a module that would add db fields and object fields

File: classes/ObjectModel.php (hunk 1 above)
Comment (Shagshag) [inline classes/ObjectModel.php]: Fixed, thank you.

**Comment by Shudrum** [comment]
Hi @Shagshag!

Sorry for the long time…

The PrestaShop hooks are really  resources consuming, we cannot add them everywhere.
Here, two calls for every object for PrestaShop will considerably drop the performances of the entire application.

I'm sorry (again), but I could not merge it.

Thank you!

**Pull request closed without merging**